src/api.js

```diff
@@ -0,0 +1,110 @@
+  const entropy = options.entropy ? options.entropy.slice(0, 16) : brorand(16);
+  return addressCodec.encodeSeed(entropy, options.algorithm || 'secp256k1');
+}
+
+function hash(message) {
+  return hashjs.sha512().update(message).digest().slice(0, 32);
+}
+
+const secp256k1 = {
+  deriveKeypair: function(entropy, options) {
+    const prefix = '00';
+    const privateKey = prefix + derivePrivateKey(entropy, options)
+      .toString(16, 64).toUpperCase();
+    const publicKey = bytesToHex(Secp256k1.keyFromPrivate(
+      privateKey.slice(2)).getPublic().encodeCompressed());
+    return {privateKey, publicKey};
+  },
+  sign: function(message, privateKey) {
+    return bytesToHex(Secp256k1.sign(hash(message),
+      hexToBytes(privateKey), {canonical: true}).toDER());
+  },
+  verify: function(message, signature, publicKey) {
+    return Secp256k1.verify(hash(message), signature, hexToBytes(publicKey));
+  }
+};
+
+const ed25519 = {
+  deriveKeypair: function(entropy) {
+    const prefix = 'ED';
+    const rawPrivateKey = hash(entropy);
+    const privateKey = prefix + bytesToHex(rawPrivateKey);
+    const publicKey = prefix + bytesToHex(
+      Ed25519.keyFromSecret(rawPrivateKey).pubBytes());
+    return {privateKey, publicKey};
+  },
+  sign: function(message, privateKey) {
+    return bytesToHex(Ed25519.sign(
+      message, hexToBytes(privateKey).slice(1)).toBytes());
+  },
+  verify: function(message, signature, publicKey) {
+    return Ed25519.verify(message, hexToBytes(signature),
+      hexToBytes(publicKey).slice(1));
+  }
+};
+
+function select(algorithm) {
+  const methods = {secp256k1, ed25519};
+  return methods[algorithm];
+}
+
+function deriveKeypair(seed, options) {
+  const decoded = addressCodec.decodeSeed(seed);
+  return select(decoded.type).deriveKeypair(decoded.bytes, options);
+}
+
+function getAlgorithmFromKey(key) {
+  const bytes = hexToBytes(key);
+  return (bytes.length === 33 && bytes[0] === 0xED) ? 'ed25519' : 'secp256k1';
+}
+
+function sign(message, privateKey) {
+  const algorithm = getAlgorithmFromKey(privateKey);
+  return select(algorithm).sign(message, privateKey);
+}
+
+function verify(message, signature, publicKey) {
+  const algorithm = getAlgorithmFromKey(publicKey);
+  return select(algorithm).verify(message, signature, publicKey);
+}
+
+function deriveAddressFromBytes(publicKeyBytes) {
+  return addressCodec.encodeAccountID(
+    utils.deriveAccountIDBytes(publicKeyBytes));
+}
+
+function deriveAddress(publicKey) {
+  return deriveAddressFromBytes(hexToBytes(publicKey));
+}
+
+function deriveNodeAddress(publicKey) {
+  const generatorBytes = addressCodec.decodeNodePublic(publicKey);
+  const accountPublicBytes = accountPublicFromPublicGenerator(generatorBytes);
+  return deriveAddressFromBytes(accountPublicBytes);
+}
+
+module.exports = {
+  generateSeed,
+  deriveKeypair,
+  sign,
+  verify,
+  deriveAddress,
+  deriveNodeAddress
```

Comment:
`deriveNodeAddress` is pretty in that it fits in with the cute api, and no worse than node nodePublicAccountID.
At least in rippled there is such a thing as a NodeID which hashes the public key to produce an id in the same way as  for accounts. That hashes the public key directly, whereas this generates another public key, as if the passed in public key were the public generator in an account family.  

`address` always means AccountID in the new lexicon?